### PR TITLE
[Bug] Fix disappearance of charts after theme switch on mobile

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
Closes https://github.com/mckinsey/vizro/issues/508

Issue seems to be caused by two factors:
- Changing from grid to flex layout when using mobile
- Charts not adhering to their entire width/height after theme switch

The current CSS fix forces the areas to always take on 100% height / width that's been allocated to them.

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/changelog.d/20240603_161127_huong_li_nguyen_fix_mobile_charts.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<!--
+### Deprecated
+
+- A bullet item for the Deprecated category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Fixed
+
+- Fix disappearance of charts after theme switch on the mobile version. ([#511](https://github.com/mckinsey/vizro/pull/511))
```
Comment (antonymilne) [inline vizro-core/changelog.d/20240603_161127_huong_li_nguyen_fix_mobile_charts.md:40]:
```suggestion
- Fix disappearance of charts after theme switch on mobile. ([#511](https://github.com/mckinsey/vizro/pull/511))
```

**Review by antonymilne — APPROVED** [review]
Amazingly quick fix! ⭐ I didn't test it at all but assuming someone else does then looks great to me you for doing it 😀

**Review by petar-qb — COMMENTED** [review]
The changes look good to me, but I can't replicate the original issue.

**Review by nadijagraca — APPROVED** [review]
Looks good. 🚀

**Comment by nadijagraca** [comment]
> The changes look good to me, but I can't replicate the original issue.

Did you try to replicate it on mobile layout? I tried mobile layout in main, and charts were disappearing on theme switch.

**Comment by petar-qb** [comment]
> Did you try to replicate it on mobile layout? I tried mobile layout in main, and charts were disappearing on theme switch.

Of course. I set the mobile dimensions exactly as they are in the original issue and ran the app from the main branch, but I couldn't replicate it. Anyway, if it works for you too, then let's merge this PR 😄.

**Pull request merged**